pkg/virt-launcher/virtwrap/converter/archconverter.go

```diff
@@ -31,6 +31,7 @@ const (
 
 type archConverter interface {
 	addGraphicsDevice(vmi *v1.VirtualMachineInstance, domain *api.Domain, c *ConverterContext)
+	scsiController(c *ConverterContext, driver *api.ControllerDriver) api.Controller
```

Comment:
While it's not necessarily ground for holding the PR further, I really feel that the central commit does way, way too much stuff all at once.

This one here is a perfect example: I would much rather have a small commit that introduces the `scsiController()` (should it be `createScsiController()` instead?) API and converts existing architectures to it as preparation, instead of seeing this refactoring thrown in with everything else.

Similarly, test suite changes could be their own commit, as should be the fairly sizeable changes to how usbredir support gets built.

I realize that by this point you're probably completely sick of looking at this code and just want to be done with it :) so I won't push too hard on this front. But I couldn't avoid voicing this concern.